### ISSUE TITLE
[HttpStress] [SslStress] Windows stress builds are broken

### ISSUE THREAD

**Issue by antonfirsov** [dotnet/runtime#95750]
```
2023-12-07T13:23:15.1583259Z   D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): error C3611: 'System::Numerics::INumberBase::System::IUtf8SpanFormattable.TryFormat': a sealed function cannot have a pure-specifier
```

The failures started on the 17th of November, across all branches, which indicates an image change might have triggered this, we should do a comparison similar to https://github.com/dotnet/runtime/issues/82528#issuecomment-1446676611.

*Edit: links to the pipelines*
- [http](https://github.com/dotnet/runtime/blob/8b04e76468af5905f739506e97039170b7554671/eng/pipelines/libraries/stress/http.yml#L111-L113)
- [ssl](https://github.com/dotnet/runtime/blob/8b04e76468af5905f739506e97039170b7554671/eng/pipelines/libraries/stress/ssl.yml#L63-L65)

*The failing build that runs **outside** of docker:* https://github.com/dotnet/runtime/blob/8b04e76468af5905f739506e97039170b7554671/eng/docker/build-docker-sdk.ps1#L24

**Comment by rzikm** [comment]
I encountered this on my dev machine. The VS installation needs to be updated, see https://github.com/dotnet/runtime/issues/94998#issuecomment-1819449017

**Comment by rzikm** [comment]
The image seems to contain the right msvc version, but the build script for some reason does not choose the latest toolchain

```
Dumping present MSVC versions


    Directory: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC


Mode                 LastWriteTime         Length Name                                                                 
----                 -------------         ------ ----                                                                 
d-----          2/4/2024  10:26 PM                14.16.27023                                                          
d-----          2/4/2024  10:27 PM                14.29.30133                                                          
d-----          2/4/2024  10:31 PM                14.37.32822                                                          
d-----          2/4/2024  10:09 PM                14.38.33130                        
```

```
 -- The CXX compiler identification is MSVC 19.37.32826.1
```

**Comment by antonfirsov** [comment]
I wonder what will happen if you manually delete the `<=14.37.32822` directories?

**Comment by rzikm** [comment]
> I wonder what will happen if you manually delete the `<=14.37.32822` directories?

then it tries to use GCC 🤦

https://dev.azure.com/dnceng-public/public/_build/results?buildId=608756&view=logs&j=0bc77094-9fcd-5c38-f6e4-27d2ae131589&t=4fd93899-d50b-5dc1-f9eb-66ab1b883c22&l=704

**Comment by rzikm** [comment]
Okay, I think I have spent enough time on this and am just trying at random, I think we need help from @dotnet/dnceng to debug this. Can you please take a look?

**Comment by markwilkie** [comment]
@rzikm - what are you hoping dnceng can help with more specifically?

**Comment by rzikm** [comment]
we are running the sslstress pipeline on `1es-windows-2022-open` images, as such, we need first to build the .NET runtime and then build the stress application against it and run it in docker.

Since November, the build has been failing due to https://github.com/dotnet/runtime/issues/94998, the fix for that was supposed to be updating Visual Studio installation.

According to the [list of included software of the image](https://github.com/actions/runner-images/blob/win22/20240204.1/images/windows/Windows2022-Readme.md), the new toolchain is installed and it indeed shows up in the relevant directories

The image seems to contain the right msvc version.

```
    Directory: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC


Mode                 LastWriteTime         Length Name                                                                 
----                 -------------         ------ ----                                                                 
d-----          2/4/2024  10:26 PM                14.16.27023                                                          
d-----          2/4/2024  10:27 PM                14.29.30133                                                          
d-----          2/4/2024  10:31 PM                14.37.32822                                                          
d-----          2/4/2024  10:09 PM                14.38.33130                        
```

However, the build still ends up using the older toolchain, and thus failing to build

```
 -- The CXX compiler identification is MSVC 19.37.32826.1
```

The failing step is simply building the SDK docker image via the `build-docker-sdk.ps1 -w ...`.

> @rzikm - what are you hoping dnceng can help with more specifically?

We are hoping that dnceng would help us get the build to work, since we are using the "official" script to build the image.

(note that the .net runtime is being built on the host, not inside docker image due to https://github.com/dotnet/runtime/pull/700)

**Comment by markwilkie** [comment]
You had a chance to check out https://github.com/dotnet/arcade/blob/main/Documentation/ArcadeSdk.md ?  It might shed more light on why your build is picking up the wrong toolchain.

**Comment by antonfirsov** [comment]
@markwilkie this seems to be a bug in the build infra that can be reproduced with a very simple yaml, opened #100046.

**Comment by karelz** [comment]
Fixed in main (9.0) in PR #100145 and in 8.0.x in PR #100303.
